python/cudf_polars/tests/test_join.py

```diff
@@ -71,7 +72,7 @@ def test_non_coalesce_join(left, right, how, nulls_equal, join_expr):
     query = left.join(
         right, on=join_expr, how=how, nulls_equal=nulls_equal, coalesce=False
     )
-    assert_gpu_result_equal(query, check_row_order=how == "left")
+    assert_gpu_result_equal(query, check_row_order=False)
```

Comment:
Here, and in many of the other join tests, we no longer check row order.

We could make this slightly more complicated like `check_row_order=(how == "left" and BLOCKSIZE_MODE="default")`. My understanding is that newer versions of polars don't even guarantee the row order for `left` joins so I opted to just not check that.

Reply:
Yes, that's a good decision. We should also remove the "by default reorder" for left joins if we don't already.

Reply:
Looks like @Matt711 is handling that at https://github.com/rapidsai/cudf/pull/17698/files#diff-58a9bfc6f3a54d841c1a798c95354d72b05555be355adec331ca5d5d1c67ad16R1260 in the PR adding support for `maintain_order`.